src/exception.jl

```diff
@@ -44,6 +49,28 @@ function show(io::IO, e::PyError)
     end
 end
 
+function pyerr_construct(msg::AbstractString)
+        ptype, pvalue, ptraceback = Ref{PyPtr}(), Ref{PyPtr}(), Ref{PyPtr}()
+        # equivalent of passing C pointers &exc[1], &exc[2], &exc[3]:
+        ccall((@pysym :PyErr_Fetch), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        ccall((@pysym :PyErr_NormalizeException), Cvoid, (Ref{PyPtr},Ref{PyPtr},Ref{PyPtr}), ptype, pvalue, ptraceback)
+        pyerr_construct(msg, PyObject(ptype[]), PyObject(pvalue[]), PyObject(ptraceback[]))
+end
+
+function pyerr_construct(msg::AbstractString, e::PyError)
+    pyerr_construct(msg, e.T, e.val, e.traceback)
+end
+
+function pyerr_construct(msg::AbstractString, ptype, pvalue, ptraceback)
+    args = pvalue.args
```

Comment:
Probably we can make this a bit more robust, and avoid unnecessary conversions.  (Rebase to get 926b931690fcf311df5729fc021430fbd1da12b8 for `_getproperty`)  Something like:
```jl
pargs = _getproperty(pvalue, "args")
if pargs != C_NULL
    args = PyObject(pargs)
    if length(args) > 0
        arg = PyObject(ccall((@pysym :PySequence_GetItem), PyPtr, (PyPtr,Int), args, 0))
        if is_pyjlwrap(arg)
            jarg = unsafe_pyjlwrap_to_objref(arg)
            jarg isa PyJlError && return jarg
        end
    end
end
return PyError(msg, ptype, pvalue, ptraceback)